hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPOfferService.java

```diff
@@ -783,15 +784,16 @@ assert getBlockPoolId().equals(bp) :
       break;
     case DatanodeProtocol.DNA_BALANCERBANDWIDTHUPDATE:
       LOG.info("DatanodeCommand action: DNA_BALANCERBANDWIDTHUPDATE");
-      long bandwidth =
+      long encodedBandwidth =
                  ((BalancerBandwidthCommand) cmd).getBalancerBandwidthValue();
-      if (bandwidth > 0) {
+      if (encodedBandwidth > 0) {
```

Comment:
nitpick: this will always be > 0 as we now have high bits set for the command

Reply:
It might be possible for the namenode to send the 0 value outside of using the command, I did not dig entirely but I'd better not change this behavior